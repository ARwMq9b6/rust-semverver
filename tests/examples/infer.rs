pub mod old {
    mod a {
        pub trait Abc {

        }

        pub struct Klm;
    }

    pub fn a<A: a::Abc>(a: A) -> A {
        a
    }

    pub fn b() -> a::Klm {
        a::Klm
    }

    pub struct Hij {
        pub field: Box<a::Abc>,
    }
}

pub mod new {
    mod a {
        pub trait Def {

        }

        pub struct Opq;
    }

    pub fn a<A: a::Def>(a: A) -> A {
        a
    }

    pub fn b() -> a::Opq {
        a::Opq
    }

    pub struct Hij {
        pub field: Box<a::Def>,
    }
}
