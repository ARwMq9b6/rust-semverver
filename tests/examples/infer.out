version bump: 1.0.0 -> (Breaking) -> 2.0.0
warning: Breaking changes in `a`
  --> $REPO_PATH/tests/examples/infer.rs:33:5
   |
33 | /     pub fn a<A: a::Def>(a: A) -> A {
34 | |         a
35 | |     }
   | |_____^
   |
   = warning: Breaking (TypeChanged { error: Sorts(ExpectedFound { expected: fn(A) -> A {oldandnew::old::a::<A>}, found: fn(A) -> A {oldandnew::new::a::<A>} }) })

warning: Breaking changes in `b`
  --> $REPO_PATH/tests/examples/infer.rs:37:5
   |
37 | /     pub fn b() -> a::Opq {
38 | |         a::Opq
39 | |     }
   | |_____^
   |
   = warning: Breaking (TypeChanged { error: Sorts(ExpectedFound { expected: fn() -> oldandnew::old::a::Klm {oldandnew::old::b}, found: fn() -> oldandnew::new::a::Opq {oldandnew::new::b} }) })

